### PR TITLE
Add linting rule

### PR DESCRIPTION
**Pull request by zhiyuanliang-ms**
## Why this PR?

Add code format check

## Visible changes

Linting rules are defined in .editorconfig file. 

<EnforceCodeStyleInBuild> property is added to the csproj file and then all code format issues will be reported during build process. In the editorconfig file, it sets the severity of IDE0055 and some other formating rules to error.
This will make build fail if the formating rule is not followed.

### CODE REVIEW

File: tests/Tests.AzureAppConfiguration/KeyVaultReferenceTests.cs [hunk 1]
```diff
@@ -936,7 +935,7 @@ public async Task SecretsWithDifferentRefreshIntervals()
             var config = new ConfigurationBuilder()
                 .AddAzureAppConfiguration(options =>
                 {
-                    options.ClientManager = TestHelpers.CreateMockedConfigurationClientManager(mockClient.Object);;
+                    options.ClientManager = TestHelpers.CreateMockedConfigurationClientManager(mockClient.Object); ;
```
Comment (juniwang) [inline tests/Tests.AzureAppConfiguration/KeyVaultReferenceTests.cs:938]: Is there a rule to remove extra `;`?

File: tests/Tests.AzureAppConfiguration/KeyVaultReferenceTests.cs [hunk 2]
```diff
@@ -186,7 +186,7 @@ public void NotSecretIdentifierURI()
             {
                 configuration = builder.AddAzureAppConfiguration(options =>
                 {
-                    options.ClientManager = TestHelpers.CreateMockedConfigurationClientManager(mockClient.Object);;
+                    options.ClientManager = TestHelpers.CreateMockedConfigurationClientManager(mockClient.Object); ;
```
Comment (juniwang) [inline tests/Tests.AzureAppConfiguration/KeyVaultReferenceTests.cs:189]: quite a few extra `;` in this file.

File: tests/Tests.AzureAppConfiguration/KeyVaultReferenceTests.cs (hunk 1 above)
Comment (zhiyuanliang-ms) [reply tests/Tests.AzureAppConfiguration/KeyVaultReferenceTests.cs:938]: No.

File: tests/Tests.AzureAppConfiguration/KeyVaultReferenceTests.cs (hunk 2 above)
Comment (zhiyuanliang-ms) [reply tests/Tests.AzureAppConfiguration/KeyVaultReferenceTests.cs:189]: Oh, I see. I will fix them.

**Comment by juniwang** [comment]
what's the purpose of this file: https://github.com/Azure/AppConfiguration-DotnetProvider/blob/main/AzureAppConfigurationRules.ruleset ?

**Comment by zhiyuanliang-ms** [comment]
> what's the purpose of this file: https://github.com/Azure/AppConfiguration-DotnetProvider/blob/main/AzureAppConfigurationRules.ruleset ?

@juniwang This file is used for enforce [CA2007](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2007) check. #410

**Comment by juniwang** [comment]
> > what's the purpose of this file: https://github.com/Azure/AppConfiguration-DotnetProvider/blob/main/AzureAppConfigurationRules.ruleset ?
> 
> @juniwang This file is used for enforce [CA2007](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2007) check. #410
> 
> [Code analysis](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/overview?tabs=net-8) is a big topic. It includes two aspects: code quality and code style.
> 
> image
> 
> CA2007 belongs to code quality check.
> 
> Even for code style, there are 4 aspects as shown in the screenshot below.
> 
> image
> 
> In this PR, we only focus on the formatting rules.

OK,

File: examples/ConfigStoreDemo/ConfigStoreDemo.csproj [hunk 3]
```diff
@@ -1,17 +1,26 @@
...
   </ItemGroup>
+  
   <ItemGroup>
     <Content Update="appsettings.json">
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </Content>
   </ItemGroup>
+
+  <PropertyGroup>
+    <EnforceCodeStyleInBuild>True</EnforceCodeStyleInBuild>
```
Comment (juniwang) [inline examples/ConfigStoreDemo/ConfigStoreDemo.csproj:23]: There are quite a few projects in the repo. do we consider adding [Directory.props](https://learn.microsoft.com/en-us/visualstudio/msbuild/customize-by-directory?view=vs-2022#directorybuildprops-and-directorybuildtargets) to simplify the settings? and to make sure it's enabled by default when we add new projects in future.

Comment (zhiyuanliang-ms) [reply examples/ConfigStoreDemo/ConfigStoreDemo.csproj:23]: Sounds good.

**Review by amerjusupovic — APPROVED** [review]

**Pull request merged**